docs/en/04_Changelogs/4.0.0.md

```diff
@@ -12,6 +12,27 @@
 
 ## Upgrading
 
+### New asset storage mechanism
+
+File system has been abstracted into an abstract interface. By default, the out of the box filesystem
+uses [Flysystem](http://flysystem.thephpleague.com/) with a local storage mechanism (under the assets directory).
+
+Because the filesystem now uses the sha1 of file contents in order to version multiple versions under the same
+filename, the default storage paths in 4.0 will not be the same as in 3.
+
+In order to retain existing file paths in line with framework version 3 you should set the
+`\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore.legacy_paths` config to true.
```

Comment:
When 4.0 is released embedded files should work as they did before, although the new DBFile will get its own shortcode handler.